.cbmc-batch/stubs/aws_byte_buf_write_stub.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <aws/common/array_list.h>
+#include <aws/common/byte_order.h>
+#include <aws/common/common.h>
+
+#include <string.h>
+
+/* We can't include byte_buf.h, because that defines this as a static inline function, and we need to override it.
+ * So just include the struct definition we need */
+struct aws_byte_buf {
+    /* do not reorder this, this struct lines up nicely with windows buffer structures--saving us allocations.*/
+    size_t len;
+    uint8_t *buffer;
+    size_t capacity;
+    struct aws_allocator *allocator;
+};
+
+bool aws_byte_buf_is_valid(const struct aws_byte_buf *const buf);
+
+/**
+ * Stub for aws_byte_buf_write (the weird name is due to CBMC name mangling)
+ * Has the same behaviour as the actual function except id doesn't actually write the bytes.
+ * In order to use this function safely, the user must check that the byte_buf bytes are not actually
+ * used by the function under test.
```

Comment:
This sounds scary.  Can you test in the stub that the user is invoking the function safely?

Reply:
See below.  I'd like to, but not sure how given current CBMC.  We can ask CBMC team to add this as a feature in our next diffblue meeting